test/CppShared/TestMany.cpp

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include <SampleFlatC.h>
+#include <winrt/Microsoft.ProjectReunion.h>
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+
+namespace ProjectReunionCppTest
+{
+	TEST_CLASS(FlatApiTests)
+	{
+	public:
+		
+		TEST_METHOD(Creates)
+		{
+            wil::unique_sampleflat_handle temp;
+            Assert::IsTrue(SUCCEEDED(::SampleFlatApiCreate(L"kittens", &temp)));
+            Assert::IsTrue(SUCCEEDED(::SampleFlatApiCreate(L"puppies", &temp)));
+            Assert::IsTrue(FAILED(::SampleFlatApiCreate(nullptr, &temp)));
+        }
+
+        TEST_METHOD(CloseNullOk)
+        {
+            CloseSampleFlatApi(nullptr);
+        }
+
+        TEST_METHOD(StringRoundTrip)
+        {
+            const wchar_t* tasty = L"muffins";
```

Comment:
can we make wstring_view (and the L"muffins"sv pattern for literals) more of the default in this codebase? Raw pointers should only be acceptable for in parameters (and even then it would be great to recommend GSL wrappers to help).

Reply:
Sure, sounds like a great policy for the implementation. This test code is using the ABI of the DLL, which has to deal in `const wchar_t*`.